### PR TITLE
Test for wasm32/WASI on CI

### PR DESCRIPTION
**Pull request by MaxDesiatov**
I'm not 100% sure why some of the values produced by tests slightly deviate, maybe because `Double` on wasm32 is 32-bit, so there's some loss of precision?

### CODE REVIEW

File: Tests/MetadataTests.swift [hunk 1]
```diff
@@ -6,6 +6,7 @@
 //  Copyright © 2019 Nick Lockwood. All rights reserved.
 //
 
+#if !os(WASI)
```
Comment (MaxDesiatov) [inline Tests/MetadataTests.swift:9]: Disabled due to the lack of filesystem support on WASI.

**Comment by nicklockwood** [comment]
This is awesome, thanks @MaxDesiatov!

**Comment by MaxDesiatov** [comment]
Looks like SwiftWasm 5.4/5.5 CI is failing due to the lack of support for `NSKeyedUnarchiver` in those versions, I've removed them from the matrix.

**Comment by nicklockwood** [comment]
> Double on wasm32 is 32-bit

Seriously?!

**Comment by MaxDesiatov** [comment]
My bad, `Double` is 64-bit. I thought of `Int`, which indeed is 32-bit on wasm32. I don't how any good explanation then. I haven't yet exercised any of the code exposed by these tests that behaving differently, so can't say how big of an impact that is. Would have to debug thoroughly to say more. Let me know if these naive `#if os(wasm32)` checks in tests are blockers for approving and merging this, I'll prioritise debugging it in that case.

**Comment by nicklockwood** [comment]
> I don't have any other possible explanations then

The CSG stuff is pretty sensitive to slight differences in rounding, so I'm not terribly surprised by it tbh. I wouldn't worry about it for now in any case.

**Pull request merged**